lib/ansiblelint/utils.py

```diff
@@ -732,3 +739,89 @@ def get_rule_skips_from_line(line):
         noqa_text = line.split('# noqa')[1]
         rule_id_list = noqa_text.split()
     return rule_id_list
+
+
+def is_playbook(filename):
+    try:
+        f = parse_yaml_from_file(filename)
+        if isinstance(f, AnsibleSequence) and hasattr(f[0], 'hosts'):
+            return True
+    except Exception as e:
+        print(
+            "Warning: Failed to load %s with %s, assuming is not a playbook."
+            % (filename, e))
+    return False
+
+
+def get_playbooks_and_roles(verbose=False):
+
+    # git is preferred as it also considers .gitignore
+    files = OrderedDict.fromkeys(sorted(subprocess.check_output(
+        ["git", "ls-files", "*.yaml", "*.yml"],
+        universal_newlines=True).split()))
+
+    playbooks = []
+    role_dirs = []
+    role_internals = set(['defaults', 'vars', 'tasks', 'handlers', 'meta', 'files', 'templates'])
```

Comment:
no need to create an intermediate in-memory list.
```suggestion
    role_internals = {'defaults', 'vars', 'tasks', 'handlers', 'meta', 'files', 'templates'}
```

Reply:
done